privly.safariextension/scripts/content_scripts/privly.js

```diff
@@ -395,6 +400,14 @@ var privly = {
 
     //put the iframe into the page
     object.parentNode.insertBefore(iFrame, object);
+
+    // Give the iframe 2s to load. If CSP blocks iframe injection then,
+    // stop the content script.
+    setTimeout(function() {
+      if (privly.blockedByCSP === true) {
+        privly.stop();
+      }
+    }, 2000);
```

Comment:
Does this mean that we need to first have the source of the iframe as `CSP_iframe.html` and if that loads successfully then, change the source of the iframe to the original privly link?